servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/DefaultNettyConnection.java

```diff
@@ -459,37 +464,34 @@ public FlushStrategy defaultFlushStrategy() {
         void channelWritable();
 
         /**
-         * Close the channel after the pending writes complete.
-         *
+         * Notification that the channel's outbound side has been closed and will no longer accept writes.s
          * <p>
-         * Calling {@link #close(Throwable)} after {@link #closeGracefully()} will be ignored.
-         * <p>
-         * This event is expected be called from the eventloop.
+         * Always called from the event loop thread.
          */
-        void closeGracefully();
+        void channelOutboundClosed();
 
         /**
          * Notification that the channel has been closed.
          * <p>
-         * This may not always be called from the event loop. For example if the channel is closed when a new write
-         * happens then this method will be called from the writer thread.
+         * This may not always be called from the event loop thread. For example if the channel is closed when a new
+         * write happens then this method will be called from the writer thread.
          *
          * @param closedException the exception which describes the close rational.
          */
-        void close(Throwable closedException);
+        void channelClosed(Throwable closedException);
     }
 
-    private static final class NoopWritableListener implements WritableListener {
+    private static final class NoopChannelOutboundListener implements ChannelOutboundListener {
         @Override
         public void channelWritable() {
         }
 
         @Override
-        public void closeGracefully() {
+        public void channelOutboundClosed() {
```

Comment:
(I meant to link the related discussion https://github.com/apple/servicetalk/pull/919#discussion_r366569972 here)

IMHO the interface naming (prior to this PR) has confusing semantics. From a functional perspective this interface provides methods that are called when certain channel/protocol events occur (e.g. writability change, channel close, protocol write close, etc...). However the method names are targeted toward how the implementations are assumed to react. The `closeGracefully` creates confusion IMHO because it isn't clear how implementations are expected to change their internal state related to any writes/events (which maybe concurrent) that may occur (e.g. found myself asking these types of questions looking at WSS in isolation).

I see how `channelOutboundClosed` maybe ambiguous between "transport closure" and "protocol not accepting any more writes" and happy to consider renaming, but I think it makes more sense to target the event name (rather than the assumed action this event results in). What about `channelProtocolOutboundClosed` or `protocolOutboundClosed `?

Reply:
@NiteshKant - wdyt?

Reply:
I am not convinced that the method name should be targetted to the event name that we know about today (assuming we will not have any other case in which we need similar behavior, i.e. to wait for all writes to finish before closure) but this is an internal class and the naming is more about us to understand behavior. So, if you are convinced, lets go ahead.